### PR TITLE
Add admin protocol for DAC

### LINKED ISSUE #108: Implement admin protocol for dedicated administrator connection
**Is your feature request related to a problem? Please describe.**
To be more compatible with ODBC and ADO and to enable `sqlcmd -A`, the driver needs to support finding the DAC port using the browser service. 

**Describe the solution you'd like**
Implement a new protocol package for `admin` , or supplement the existing `tcp` protocol package. The new behavior will be to look up the DAC port using the browser service locally.

See https://github.com/microsoft/go-sqlcmd/issues/200

### PR DESCRIPTION
**Pull request by shueybubbles**
Fixes #108 
I've attempted to adhere to this spec for DAC: https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/diagnostic-connection-for-database-administrators?view=sql-server-ver15

The doc confuses me a bit. I don't know how an app is supposed to tell the difference between `admin:<instance_name>` and `admin:<server_name>`
The driver requires the full `admin:server\instance` syntax or separate parameters for `server` and `protocol`.

### CODE REVIEW

**Comment by shueybubbles** [comment]
Note that I added a test but a full test run against the `admin` protocol isn't (yet) feasible because the tests keep too many connections open and DAC only allows 1 open connection.

File: queries_go110_test.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build go1.10
```
Comment (stuartpa) [inline queries_go110_test.go:1]: Should we remove one of these? #Resolved

**Review by stuartpa — APPROVED** [review]
:shipit:

File: queries_go110_test.go (hunk 1 above)
Comment (shueybubbles) [reply queries_go110_test.go:1]: no, the go tooling auto-injected the second one

**Pull request merged**